doc/release/upcoming_changes/25138.c_api_removal.rst

```diff
@@ -0,0 +1,7 @@
+* The advanced indexing ``MapIter`` and related API has been removed.
+  The (truly) public part of it was not well tested and had only one
+  known user (Theano).  Making it private will simplify improvements
+  to speed up ``ufunc.at`` and make advanced indexing more maintainable
+  and was important for allowing 64 dimensional arrays.
+  Please let us know if this API is important to you so we can find a
+  solution together.
```

Comment:
Does this blurb need a title line?

Reply:
No it [does not](https://output.circle-artifacts.com/output/job/c3dde236-55f3-4d85-a0e2-a18067d294ad/artifacts/0/doc/build/html/release/2.0.0-notes.html#numpy-2-0-c-api-removals)